### PR TITLE
Fix roborio duplicate .so's on deploy

### PR DESCRIPTION
**Pull request by mcm001**
TLDR on the current problem:
photontargeting-java includes .so's
But we deploy them to the robot via jniDependencies: photontargeting-cpp
So we need to strip them from the JAR we ship to the robot

Also fixes niLibVersion.

Tested on Chris's rio! Java:
image
C++: 
image

### CODE REVIEW

File: photon-core/src/main/java/org/photonvision/common/dataflow/networktables/TimeSyncManager.java [hunk 1]
```diff
@@ -128,9 +128,10 @@ synchronized void tick() {
         if (m_client != null) {
             var conns = ntInstance.getConnections();
 
-            if (conns.length > 0) {
-                logger.debug("Changing TimeSyncClient server to " + conns[0].remote_ip);
-                m_client.setServer(conns[0].remote_ip);
+            var newServer = conns[0].remote_ip;
+            if (conns.length > 0 && !m_client.getServer().equals(newServer)) {
```
Comment (mdurrani808) [inline photon-core/src/main/java/org/photonvision/common/dataflow/networktables/TimeSyncManager.java:132]: slightly confused / im bad at programming, why do we get the remote ip from the first element than check if there are things in the list? is this just guarenteed to have one element?

File: photon-targeting/build.gradle [hunk 2]
```diff
@@ -191,6 +189,27 @@ model {
 
 apply from: "${rootDir}/shared/javacpp/publish.gradle"
 
+// quickly hack our raw jar into publishing
+def rawjavaJar = tasks.register("rawjavaJar", Jar) {
```
Comment (mdurrani808) [inline photon-targeting/build.gradle:193]: purpose?

File: photon-core/src/main/java/org/photonvision/common/dataflow/networktables/TimeSyncManager.java (hunk 1 above)
Comment (mcm001) [reply photon-core/src/main/java/org/photonvision/common/dataflow/networktables/TimeSyncManager.java:132]: totally bjork. a bug.

File: photon-targeting/build.gradle (hunk 2 above)
Comment (mcm001) [reply photon-targeting/build.gradle:193]: https://discord.com/channels/176186766946992128/368993897495527424/1306457242979139584

Comment (mcm001) [reply photon-targeting/build.gradle:193]: TLDR:
photontargeting-java includes .so's
But we deploy them to the robot via jniDependencies: photontargeting-cpp
So we need to strip them from the JAR we ship to the robot

**Review by mdurrani808 — APPROVED** [review]

**Pull request merged**